src/CMakeLists.txt

```diff
@@ -339,9 +339,10 @@ if (NN_STATIC_LIB)
     add_library (${PROJECT_NAME} STATIC ${NN_SOURCES})
 else ()
     add_library (${PROJECT_NAME} SHARED ${NN_SOURCES})
-    add_definitions (-DNN_SHARED_LIB)
     set_target_properties (${PROJECT_NAME} PROPERTIES
         SOVERSION "${NN_ABI_VERSION}")
+    set_target_properties (${PROJECT_NAME} PROPERTIES
+        COMPILE_DEFINITIONS "NN_SHARED_LIB")
```

Comment:
So this CMake thing is a bit different -- I'm concerned that if this doesn't work, we might wind up with a static library.   I *think* what you're trying to achieve here is correct, but I want to validate this.  Unfortunately, the failure mode for this if you got it wrong is that tests will just silently succeed.

Reply:
@gdamore anyway this is correct.
NN_SHARED_LIB required only for building the library. 
So I reduced the scope of compile definition for the specific target.